lib/dry/view/controller.rb

```diff
@@ -91,16 +102,41 @@ def layout?
         !!config.layout
       end
 
-      def layout_scope(renderer, context: config.context, **)
-        scope(renderer.chdir(layout_dir), EMPTY_LOCALS, context)
+      def layout_scope(renderer, context)
+        scope(renderer.chdir(layout_dir), context)
+      end
+
+      def template_scope(renderer, context, **input)
+        scope(renderer.chdir(template_path), context, locals(**input))
       end
 
-      def template_scope(renderer, context: config.context, **input)
-        scope(renderer.chdir(template_path), locals(**input), context)
+      def scope(renderer, context, locals = EMPTY_LOCALS)
+        Scope.new(
+          renderer: renderer,
+          context: context,
+          locals: decorated_locals(renderer, context, locals)
+        )
       end
 
-      def scope(renderer, locals, context)
-        Scope.new(renderer, locals, context)
+      def decorated_locals(renderer, context, locals)
+        decorator = self.class.config.decorator
+
+        locals.each_with_object({}) { |(key, val), result|
+          # Decorate truthy values only
+          if val
+            options = exposures.key?(key) ? exposures[key].options : {}
```

Comment:
I *think* `options = exposures[key].options if exposures.key?(key)` should be enough because `**nil` will work too